### PR TITLE
support android armv7a on configure script

### PR DESCRIPTION
**Pull request by EduardoRFS**
This PR adds the android triplet for armv7 which is called armv7a as a valid architecture and eabi. Essentially `androideabi` is like `gnueabi` or `musleabi` but using Android Bionic instead.

## Example

The CI below is using the same commit but on top of 4.10

https://github.com/EduardoRFS/reason-mobile/runs/2505947071

### CODE REVIEW

**Comment by EduardoRFS** [comment]
Maybe a patch like this would also be desirable?

https://github.com/EduardoRFS/ocaml/commit/aa6105cf152a08751fbdbf5e28ae8ebc0256450b

Instead of adding just this specific, this patch also supports musleabi, musleabihf and etc...

File: configure.ac [hunk 1]
```diff
@@ -979,6 +979,8 @@ AS_CASE([$host],
     [arch=arm; model=armv8; system=linux_eabi],
   [armv7*-*-linux-gnueabi],
     [arch=arm; model=armv7; system=linux_eabi],
+  [armv7a-*-linux-androideabi*],
```
Comment (dra27) [inline configure.ac:982]: Should this more pedantically be:
```suggestion
  [armv7a-*-linux-androideabi-*],
```
or is there something else which can be suffixed?

**Review by dra27 — COMMENTED** [review]
This seems fine to me, apart from a question about one `*`! For the other patch you suggest, I think wildcards are fine, but should emit a warning. So picking just this one:

```autoconf
  [armv7*-*-linux-gnueabi*],
    [arch=arm; model=armv7; system=linux_eabi],
  [armv7a-*-linux-androideabi*],
    [arch=arm; model=armv7; system=linux_eabi],
  [armv7*-*-linux-*eabi],
    [AC_MSG_WARN([Unrecognised $host; assuming armv7 linux_eabi])
    arch=arm; model=armv7; system=linux_eabi]
```

**Comment by dra27** [comment]
(I've set the no change label, but if you want to push a Changes entry, that's fine)

**Comment by Octachron** [comment]
cc @shindere who might be interested in the changes too.

File: configure.ac (hunk 1 above)
Comment (EduardoRFS) [reply configure.ac:982]: The triplet can be something like `armv7a-unknown-linux-androideabi24` so this patch would break it

Comment (dra27) [reply configure.ac:982]: Gotcha, thanks!

**Comment by EduardoRFS** [comment]
@dra27 that's a nice suggestion but I wonder if there is any environment that will match linux + eabi and not be, linux_eabi.

**Comment by dra27** [comment]
I'd wonder, too - but that's the reason to have a defensive warning! Let's see what @shindere thinks.

**Comment by shindere** [comment]
Yes, thanks, will review ASAP. Sorry for the delay!

**Comment by shindere** [comment]
David Allsopp (2021/05/05 11:34 -0700):
> I'd wonder, too - but that's the reason to have a defensive warning!
> Let's see what @shindere thinks.

Nothing special, sorry :(

**Comment by dra27** [comment]
No problem, @shindere! In which, this just needs rebasing w.r.t. #10471, if you're happy to do that @EduardoRFS? I think the case should remain distinct as it was before (i.e. #10471 has allowed `*eabi`, but because android wants `androideabi*` I think it's better to keep the specific `android` part and so a separate check).

It's up to you if you want to put the defensive warning I suggested.

**Comment by shindere** [comment]
So what's the big picture of how this will be used to produce Android
apps?

**Comment by EduardoRFS** [comment]
@shindere yeah but just binaries running on Android is quite useful, Android allows you to run full ELF binaries.

But #10471 is exactly what I was expecting, so I don't see any need for this PR more. Actually it's exactly the same patch as I did for my project at https://github.com/EduardoRFS/ocaml/commit/aa6105cf152a08751fbdbf5e28ae8ebc0256450b

**Comment by EduardoRFS** [comment]
In fact I have a couple of examples of running OCaml on Android and iOS quite nicely with the patch at #10471

**Pull request closed without merging**